src/services/database.ts

```diff
@@ -36,90 +35,58 @@ interface UpdatedPageDocument {
 
 export type PageDocType = PageDocument | UpdatedPageDocument;
 
-const ATLAS_URI = process.env.ATLAS_URI || '';
 const METADATA_COLLECTION = 'metadata';
 const PAGES_COLLECTION = 'documents';
 const UPDATED_PAGES_COLLECTION = 'updated_documents';
 const ASSETS_COLLECTION = 'assets';
 
-const logger = initiateLogger();
-
-let client: MongoClient;
-let dbInstance: Db;
+let db: Db;
 
 const getPageIdQuery = (projectName: string, branch: string) => {
   const user = process.env.BUILDER_USER ?? 'docsworker-xlarge';
   const pageIdPrefix = `${projectName}/${user}/${branch}`;
   return { $regex: new RegExp(`^${pageIdPrefix}/`) };
 };
 
-// Set up MongoClient for application
-export const setupClient = async (mongoClient: MongoClient) => {
-  client = mongoClient;
-  await client.connect();
-  const dbName = process.env.SNOOTY_DB_NAME || 'snooty_dev';
-  dbInstance = client.db(dbName);
-};
-
-// Sets up the MongoClient and returns the newly created db instance, if they don't
-// already exist
-export const db = async () => {
-  if (!dbInstance) {
-    try {
-      await setupClient(new MongoClient(ATLAS_URI));
-    } catch (e) {
-      logger.error(e);
-      throw e;
-    }
-  }
-  return dbInstance;
-};
-
-export const closeDBConnection = async () => {
-  if (client) {
-    await client.close();
-    logger.info('MongoDB Client closed successfully');
-  }
+// sets module's db scope
+// creates a new db instance, if it doesn't already exist
+export const initDb = async (client: MongoClient) => {
```

Comment:
I don't think we need to keep this async anymore

Reply:
good catch!